### PR TITLE
Elasticsearch: revert to isoWeek when resolving weekly indices

### LINKED ISSUE #31634: Elasticsearch weekly index pattern broken
**What happened**:
Used index pattern: `[logging-something*-]GGGG.WW`
Users with a browser local of en-US do not see the recent data / data of this week.

**How to reproduce it (as minimally and precisely as possible)**:
Switching the browser locale to en-GB fixes the issue.

**Anything else we need to know?**:
https://github.com/grafana/grafana/pull/29841/files#diff-56a4a2354a8d6139680ca2c9727bec82efb5ffd660ee1dc66b5508bbe8c0ce06R15

Looks like there was a switch away from ISO weeks in #29841. US and Europe count the weeks differently now.

**Environment**:
- Grafana version: 7.4.3
- Data source type & version: ES 7.9.x

### PR DESCRIPTION
**Pull request by Elfo404**
**What this PR does / why we need it**:

As described in the issue, in https://github.com/grafana/grafana/pull/29841 we mistakenly switched from `isoWeek` to `week` when resolving weekly indices in elasticsearch. This reverts the change back.

**Which issue(s) this PR fixes**:

Fixes #31634

### CODE REVIEW

File: packages/grafana-data/src/datetime/moment_wrapper.ts [hunk 1]
```diff
@@ -96,7 +97,7 @@ export const toUtc = (input?: DateTimeInput, formatInput?: FormatInput): DateTim
 };
 
 export const toDuration = (input?: DurationInput, unit?: DurationUnit): DateTimeDuration => {
-  return moment.duration(input as DurationInputArg1, unit) as DateTimeDuration;
+  return moment.duration(input as DurationInputArg1, unit as DurationInputArg2) as DateTimeDuration;
```
Comment (Elfo404) [inline packages/grafana-data/src/datetime/moment_wrapper.ts:100]: seems like moment types don't include `isoWeek`, but it's actually present in the source, that's also what we were using before strongly typing https://github.com/grafana/grafana/pull/31709/files#diff-56a4a2354a8d6139680ca2c9727bec82efb5ffd660ee1dc66b5508bbe8c0ce06R12 not sure we have a better way of fixing this.

Comment (aocenas) [reply packages/grafana-data/src/datetime/moment_wrapper.ts:100]: Would put this as a comment to explain the type cast here.

**Review by aocenas — APPROVED** [review]

File: packages/grafana-data/src/datetime/moment_wrapper.ts (hunk 1 above)
Comment (Elfo404) [reply packages/grafana-data/src/datetime/moment_wrapper.ts:101]: good point, thanks!

**Review by hugohaggmark — APPROVED** [review]
Looks good 🎉 !

**Pull request merged**